DEVELOPER_GUIDE.md

```diff
@@ -422,6 +423,79 @@ Output plugins are defined in [flb_output.h](https://github.com/fluent/fluent-bi
+
+There are various types of supported configuration types. Full list available [here](https://github.com/fluent/fluent-bit/blob/master/include/fluent-bit/flb_config_map.h#L29). The most used ones are :
+
+| Type                   | Description           | 
+| -----------------------|:---------------------:| 
+| FLB_CONFIG_MAP_INT     | Represents integer data type | 
+| FLB_CONFIG_MAP_BOOL    | Represents boolean data type | 
+| FLB_CONFIG_MAP_DOUBLE  | Represents a double |
+| FLB_CONFIG_MAP_SIZE    | Converts string size to integer |
+| FLB_CONFIG_MAP_STR     | Represents string data type |
+| FLB_CONFIG_MAP_CLIST   | Comma separated list of strings |
+| FLB_CONFIG_MAP_SLIST   | Empty space separated list of strings |
+
+A config map expects certain public fields at registration. They are 
+
+| Public Fields | Description           | 
+| --------------|:---------------------| 
+| Type          | This field requires the data type of the property that we are writing to the config map. If the property is of type `int` we use `FLB_CONFIG_MAP_INT`, if `string` `FLB_CONFIG_MAP_STR` etc. |
+| Name          | This field requires the name of the configuration property. For example for the property flush count we use `flush_count`|
+| Default Value | This field allows the user to set the default value of the property.For example, for a property of type `FLB_CONFIG_MAP_BOOL` (boolean), the default value may be false. Then we have to give `false` as default value. If there is no default value, `NULL` is given.|
+| Flags         | This field allows the user to set option flags. For example, it specifies in certain cases if multiple entries are allowed. |
+| Set Property  | This field decides if the property needs to be written to context or just validated. If the property needs to be written to context, the value of this field needs to `FLB_TRUE` or else the value will be `FLB_FALSE`.|
+| Offset        | This field represents the member offset. It is 0 if the property is not written to context and if the property is being written to context it is ```offsetof(struct name_of_plugin_structure, name_of_property)```. The macro offsetof() returns the offset of the field *member* from the start of the structure type.|
+| Description   | This field is so that the user can give a short description of the property. It is `NULL` if no description is needed or given. |
+
+For example for [stdout](https://github.com/fluent/fluent-bit/blob/master/plugins/out_stdout/stdout.c#L158) plugin the config map is something like :
+
+```c
+/* Configuration properties map */
+static struct flb_config_map config_map[] = {
+    {
+     FLB_CONFIG_MAP_STR, "format", NULL,
+     0, FLB_FALSE, 0,
+     NULL
+    },
```

Comment:
I think for the dev guide, it'd be good to have a full example with descriptions for the fields. 

I recommend adding that to the example here, and then separate submitting a PR to update the stdout plugin config map code.

Reply:
https://github.com/fluent/fluent-bit/pull/2075
Pl review